app/Jobs/InstallEgg.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Services\Eggs\Sharing\EggImporterService;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Log;
+use Throwable;
+
+class InstallEgg implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public string $downloadUrl;
+
+    public int $timeout = 15;
+
+    public function __construct(string $downloadUrl)
+    {
+        $this->downloadUrl = $downloadUrl;
+    }
+
+    /**
+     * @throws Throwable
+     */
+    public function handle(EggImporterService $eggImporterService): void
+    {
+        try {
+            $eggImporterService->fromUrl($this->downloadUrl);
+        } catch (Throwable $e) {
+            Log::error('InstallEgg job failed.', [
+                'downloadUrl' => $this->downloadUrl,
+                'exception' => $e->getMessage(),
+            ]);
```

Comment:
```suggestion
            report($e);

```

Reply:
I tweaked it, reporting just the exception does not provide the download url that failed. So we have no way of knowing what failed.